.github/workflows/run-tests.yml

```diff
@@ -20,10 +20,24 @@ on:
 
 jobs:
   run-tests:
-    name: Run tests job
+    name: Run tests job matrix
 
     runs-on: ubuntu-latest
 
+    strategy:
+      matrix:
+        client-type: [go, http]
+        database-type: [badger-file, badger-memory]
+        mutation-type: [gql, collection-named, collection-save]
+        tests: [names, lens, cli]
```

Comment:
suggestion: I would remove this line, and instead all items can just test `./...`.